File: front/lib/models/agenda.dart
class Agenda {
  int? id;
  String? date;
  String? time;
  String? endtime;
  String? status;
  int? id_doctor;

  Agenda({
    this.id,
    this.date,
    this.time,
    this.endtime,
    this.status,
    this.id_doctor,
  });

  factory Agenda.fromJson(Map<String, dynamic> json) {
    return Agenda(
      id: json["id"],
      date: json["date"],
      time: json["time"],
      endtime: json["end_time"],
      status: json["status"],
      id_doctor: json["id_doctor"],
    );
  }
}

File: api/app/Models/Agenda.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use App\Models\User;


class Agenda extends Model
{
    use HasFactory;
    protected $fillable = [
        'date',
        'time',
        'end_time',
        'status',
        'id_doctor'
    ];
    public function user()
    {
        return $this->belongsTo(User::class);
    }
}
